server/controllers/donation.js

```diff
@@ -53,7 +48,8 @@ export default {
 
   hasAuthorization(req, res, next) {
     if (req.user.roles.find(r => r === ADMIN_ROLE) ||
-        req.params.donationId === req.user._id) {
+        req.params.donationId === req.user._id ||
+        req.body.donor === req.user._id) {
```

Comment:
This can cause an issue since hasAuthorization is also used in sendEmail.  A malicious user can login with any user id.  Then he can send put requests to `/donations/:donationId` with body `{ donor: (his userId)}`.  This will cause hasAuthorization to pass and will allow him to access this endpoint for any `:donationId`.  Is there another way you can do this that doesn't open up this hole?